core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -644,96 +643,81 @@ public RoleBinding getRoleBinding(String name) {
 	 * @return List of role bindings that aren't considered default.
 	 */
 	public List<RoleBinding> getUserRoleBindings() {
-		return roleBindings().withoutLabel(KEEP_LABEL).list().getItems().stream()
+		return rbac().roleBindings().withoutLabel(KEEP_LABEL).list().getItems().stream()
 				.filter(rb -> !rb.getMetadata().getName().matches("admin|system:deployers|system:image-builders|system:image-pullers"))
 				.collect(Collectors.toList());
 	}
 
 	public boolean deleteRoleBinding(RoleBinding roleBinding) {
-		return roleBindings().delete(roleBinding);
+		return rbac().roleBindings().delete(roleBinding);
 	}
 
 	public RoleBinding addRoleToUser(String roleName, String username) {
 		RoleBinding roleBinding = getOrCreateRoleBinding(roleName);
-
-		addSubjectToRoleBinding(roleBinding, "User", username);
-		addUserNameToRoleBinding(roleBinding, username);
-
+		addSubjectToRoleBinding(roleBinding, "User", username, null);
 		return updateRoleBinding(roleBinding);
 	}
 
 	public RoleBinding addRoleToServiceAccount(String roleName, String serviceAccountName) {
-		RoleBinding roleBinding = getOrCreateRoleBinding(roleName);
-
-		addSubjectToRoleBinding(roleBinding, "ServiceAccount", serviceAccountName);
-		addUserNameToRoleBinding(roleBinding, String.format("system:serviceaccount:%s:%s", getNamespace(), serviceAccountName));
+		return addRoleToServiceAccount(roleName, serviceAccountName, getNamespace());
+	}
 
+	public RoleBinding addRoleToServiceAccount(String roleName, String serviceAccountName, String namespace) {
+		RoleBinding roleBinding = getOrCreateRoleBinding(roleName);
+		addSubjectToRoleBinding(roleBinding, "ServiceAccount", serviceAccountName, namespace);
 		return updateRoleBinding(roleBinding);
 	}
 
 	public RoleBinding addRoleToGroup(String roleName, String groupName) {
 		RoleBinding roleBinding = getOrCreateRoleBinding(roleName);
-
-		addSubjectToRoleBinding(roleBinding, "SystemGroup", groupName);
-		addGroupNameToRoleBinding(roleBinding, groupName);
-
+		addSubjectToRoleBinding(roleBinding, "Group", groupName, null);
 		return updateRoleBinding(roleBinding);
 	}
 
 	private RoleBinding getOrCreateRoleBinding(String name) {
 		RoleBinding roleBinding = getRoleBinding(name);
 
 		if (roleBinding == null) {
+			// e.g `view` is ClusterRole in K8s RBAC
+			// if the roles isn't present fallback to ClusterRole
+			String kind = rbac().roles().withName(name).get() == null ? "ClusterRole" : "Role";
 			roleBinding = new RoleBindingBuilder()
 					.withNewMetadata().withName(name).endMetadata()
-					.withNewRoleRef().withName(name).endRoleRef()
+					.withNewRoleRef().withKind(kind).withName(name).endRoleRef()
```

Comment:
@Crumby imo this's ready. However, I'm not particularly sure about this "opinionated" part of code. I seemed to work in my tests, but...

Reply:
maybe make it explicit (have  addClusterRoleToServiceAccount  )

Reply:
I do agree that'd make the methods cleaner. However, if we split the API to `addRole...` and `addClusterRole...`, it'll eventually break all the usage where `view` is added to SA.